src/main/java/com/cloudbees/hudson/plugins/folder/computed/ComputedFolder.java

```diff
@@ -151,7 +151,39 @@ protected final void init() {
         for (Trigger t : triggers) {
             t.start(this, Items.currentlyUpdatingByXml());
         }
-        loadComputation();
+        synchronized (this) {
+            computation = createComputation(null);
+        }
+    }
+
+    @Override
+    public void onCreatedFromScratch() {
+        try {
+            FileUtils.forceMkdir(getComputationDir());
```

Comment:
Found a bug that the directory was being created too early within the constructor (which with mangled names gives duplicate partially populated directories - including the `config.xml`, which causes problems on reload when the child items are also folders), so when I removed that I needed to ensure the directory is created here.